Generates Dataframe JSON with supporting ETL functions

jermnelson opened the pull request:
Fixes #38, created new `src/lib/etl.py` module for extracting data from our data sources, transforming into panda DataFrames, and then loading results after processing through `nlp` pipeline for the new `species-records.json` DataFrame that will be used by the SPOC Verifier Streamlit app.

jgreben commented on src/lib/etl.py:
```diff
@@ -0,0 +1,110 @@
+TEI = {"tei": "http://www.tei-c.org/ns/1.0"}
+
+
+def get_date(xml_doc: etree.Element) -> str:
```
Does anything call this method (besides the test)?

jermnelson replied:
> Maybe a walkthrough would be in order. I don't see where some of the methods are being called. Are they called somewhere external to this repo?

@jgreben  sounds good; maybe pair after stand-up?

jgreben replied:
My apologies, I missed where this called in your walkthrough and I do not see it in data-prep. Can you just add the comment to where it is called here. Thanks!

jermnelson replied:
Done, added comment in `process_xml` function.

jgreben commented on src/lib/etl.py:
```diff
@@ -0,0 +1,110 @@
+    return species, locations, habitats
+
+
+def get_all_text(xml_doc: etree.Element) -> str:
```
Does anything call this method (besides the test)?

jermnelson replied:
No, removed from latest commit.

jgreben commented on src/lib/etl.py:
```diff
@@ -0,0 +1,110 @@
+    return records
+
+
+def process_xml(raw_xml: str, filename: str, nlp: spacy.lang) -> list:
```
Does anything call this?

jgreben replied:
Ok, I see it now in the data-prep notebook,

jgreben reviewed (commented):
Maybe a walkthrough would be in order. I don't see where some of the methods are being called. Are they called somewhere external to this repo?

jermnelson commented:
@jgreben removed function `get_all_text` along with accompanying tests. Also, added a comment to the `process_xml` function about being called from the data-prep notebook. Added new ticket #58  for testing `process_xml` and `process_div` functions.
